File: javascript/rna-transcription/rna-transcription.js
var DnaTranscriber = function () {
};

DnaTranscriber.prototype.isInvalid = function (dna) {
  if(dna.search('X') >= 0){
    return true
  } else {
    return false;
  }
}

DnaTranscriber.prototype.toRna = function (dna) {
  if(this.isInvalid(dna)){
    throw new Error('Invalid input');
  } else {
    return dna.replace(/A/g,'U')
              .replace(/T/g,'A')
              .replace(/G/g,'X')
              .replace(/C/g,'G')
              .replace(/X/g,'C');
  }
};

module.exports = DnaTranscriber;

File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand):
    # dna: A C G T
    # rna: U G C A
    # first A to U, then T to A, then G to X, C to G, X to C
    if dna_strand.count('U') > 0:
        raise ValueError('dotplus')
    if dna_strand.count('X') > 0:
        raise ValueError('dotplus')
    return dna_strand.replace('A','U').replace('T','A').replace('G','X').replace('C','G').replace('X','C')


